gazelle/python/kinds.go

```diff
@@ -97,4 +101,10 @@ var pyLoads = []rule.LoadInfo{
 			pyTestKind,
 		},
 	},
+	{
+		Name: "@aspect_rules_py//py:defs.bzl",
```

Comment:
Similarly to how we discussed the gazelle plugin being agnostic to the rule sets, it would be good to have a solution for pytest that is not specific to any specific python ruleset.

Having rules_py specific logic in a rules_python gazelle plugin sounds like our abstractions within the bazel python ecosystem are not working well enough.

We had a similar discussion about rules_pycross and rules_python targets having different naming conventions and decided to make the plugin configurable, but in this case this is overstepping that boundary and starts tailoring the gazelle plugin to very specific implementation logic.

Are there different ways to achieve the same thing?

Reply:
We could put the plugin logic in rules_py itself, and then teach users how to compile a gazelle binary that has the plugins listed in the right order. I don't think we want to grow a Go dev-dependency if we can help it though.

Reply:
We can make this file a template and allow users to pass extra stuff.

Reply:
I am concerned that adding this special one off for `py_pytest_main` we are not making orthogonal rulesets that are easy to compose and will require continuous maintenance to tweak things. Since there are other test frameworks (like `behave`, `robot`, etc), catering for test frameworks in the gazelle plugin means that we are willing to accept contributions for all, hence I think that solving this differently would be better.

I liked the solution that @linzhp proposed in #2040. It does not add extra test framework specific concepts to the gazelle plugin and it would work with `pytest_main` if the users are willing to define a macro and use it, so whilst that may require users to write more code, it does mean that the `rules_python` project stays better scoped.

We have a maintainers meeting every Monday evening US time and we could discuss it there next week if one of you joined?

Reply:
A modestly sophisticated user can discover the pattern suggested in #2044 to copy-paste a macro into their codebase and configure a `# map_kind` directive. I think that's good enough for the use cases that gazelle is trying to address.

We might do something more novice-friendly specifically in Aspect CLI.